keras/engine/topology.py

```diff
@@ -806,33 +856,78 @@ def output_shape(self):
                             'ill-defined for the layer. ' +
                             'Use `get_output_shape_at(node_index)` instead.')
 
-    def add_updates(self, updates, inputs):
+    def add_loss(self, losses, inputs=None):
+        if losses is None:
+            return
+        # Update self.losses
+        losses = to_list(losses)
+        if not hasattr(self, 'losses'):
+            self.losses = []
+        try:
```

Comment:
The right way to check if a class attribute is a property is:
```python
is_property = isinstance(type(obj).attribute, property)
```

So here it would be:

```python
if not isinstance(type(self).losses, property):
    self.losses += losses
```